tests/Unit/IO/Test_VolumeData.py

```diff
@@ -10,160 +10,135 @@
 
         if os.path.isfile(self.file_name):
             os.remove(self.file_name)
 
-        h5_file = spectre_h5.H5File(file_name=self.file_name,
-                                    append_to_file=True)
-        self.tensor_components_and_coords = [
-            ds.DataVector([8.9, 7.6, 3.9, 2.1, 18.9, 17.6, 13.9, 12.1]),
-            ds.DataVector([0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0]),
-            ds.DataVector([0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0]),
-            ds.DataVector([0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 1.0]),
-            ds.DataVector([-78.9, -7.6, -1.9, 1.9, 6.3, 1.7, 9.8, 1.2]),
-            ds.DataVector([-7.9, 17.6, 1.9, -8.1, -6.3, 21.7, 6.8, -0.2]),
-            ds.DataVector([1.0, 6.6, 28.0, -8.1, -26.3, 32.7, 26.8, -3.2])
-        ]
+        self.h5_file = spectre_h5.H5File(file_name=self.file_name,
+                                         append_to_file=True)
+        self.tensor_components_and_coords = np.random.rand(2, 8)
         observation_ids = [8435087234, 6785087234]
         observation_values = [7.0, 1.3]
-        grid_names = ["[[2,3,4]]", "[[5,6,7]]"]
+        grid_names = ["grid_1", "grid_2"]
 
         # Insert .vol file to h5 file
-        h5_file.insert_vol("/element_data", 0)
-        vol_file = h5_file.get_vol(path="/element_data")
+        self.h5_file.insert_vol("/element_data", version=0)
+        self.vol_file = self.h5_file.get_vol(path="/element_data")
+
         # Set TensorComponent and ExtentsAndTensorVolumeData to
         # be written
-        extents_and_tensor_vol_obs1 = [
+        extents_and_tensor_vol_grid_1 = [
             ds.ExtentsAndTensorVolumeData([2, 2, 2], [
-                ds.TensorComponent(grid_names[0] + "/S",
-                                   self.tensor_components_and_coords[0]),
-                ds.TensorComponent(grid_names[0] + "/x-coord",
-                                   self.tensor_components_and_coords[1]),
-                ds.TensorComponent(grid_names[0] + "/y-coord",
-                                   self.tensor_components_and_coords[2]),
-                ds.TensorComponent(grid_names[0] + "/z-coord",
-                                   self.tensor_components_and_coords[3]),
-                ds.TensorComponent(grid_names[0] + "/T_x",
-                                   self.tensor_components_and_coords[4]),
-                ds.TensorComponent(grid_names[0] + "/T_y",
-                                   self.tensor_components_and_coords[5]),
-                ds.TensorComponent(grid_names[0] + "/T_z",
-                                   self.tensor_components_and_coords[6])
+                ds.TensorComponent(
+                    grid_names[0] + "/field_1",
+                    ds.DataVector(self.tensor_components_and_coords[0])),
+                ds.TensorComponent(
+                    grid_names[0] + "/field_2",
+                    ds.DataVector(self.tensor_components_and_coords[1]))
```

Comment:
better generate 4 rows of data above so the data for the two `observation_values` doesn't have to be the same. It can be convenient for you to iterate over the `observation_values` with `for i, observation_value in enumerate(observation_values)` so you have the index for pointing into the tensor components data.